common-tasks/usb.md

```diff
@@ -21,199 +21,108 @@ redirect_from:
+[qvm-pci][Assigning Devices] command. For guidance on finding the correct USB
+controller, see [here][usb-controller].)
 
-**Caution:** Please carefully read the [Security Warning about USB Input Devices] before proceeding.
+ * R4.0
+
+USB drive mounting is integrated into the Connection Widget. This is the tooltray
+icon with a yellow square located in the top right of your screen by default.
+Simply insert
+your USB drive and click on the widget. You will see multiple entries for your
+USB drive; typically, `sys-usb:sda`, `sys-usb:sda1`, and `sys-usb:2-1` for example.
+The simplest (but slightly less secure, see note below about attaching individual
+partitions) option is to attach the entire block drive. In our example, this is `sda`,
+so hover over it.
+This will pop up a submenu showing running VMs to which the USB drive can be connected.
+Click on one and your USB drive will be attached!
+
+Note that attaching individual partitions can be slightly more secure because it doesn't
+force the target AppVM to parse the partition table. However, it often means the 
+AppVM won't detect the new partition and you will need to manually mount it inside
+the AppVM. To do this with the GUI,
+you'd select the `sda1` entry in our example and proceed to connect to an AppVM.
+Once the USB drive has been attached to the AppVM, it will
+appear as `/dev/xvd*` (usually `xvdi` but sometimes with higher letters if you
+have multiple devices attached.) Follow the below steps if you need to manually mount
+the partition:
+    ```
+    cd ~
+    mkdir mnt
+    sudo mount /dev/xvdi mnt
+    ```
+    And when done:
+    `sudo umount mnt`
+    
+The command-line tool you may use to mount whole USB drives or their partitions
+is `qvm-block`. This tool can be used to assign a USB drive to a qube as
+follows:
 
-In order to use a USB keyboard, you must first attach it to a USB qube, then give that qube permission to pass keyboard input to dom0.
-Edit the `qubes.InputKeyboard` policy file in dom0, which is located here:
+ 1. Insert your USB drive.
 
-    /etc/qubes-rpc/policy/qubes.InputKeyboard
+ 2. In a dom0 console (running as a normal user), list all available block
+    devices:
 
-Add a line like this one to the top of the file:
+        qvm-block
 
-    sys-usb dom0 allow,user=root
+    This will list all available block devices connected to any USB controller
+    in your system, no matter which qube hosts the controller. The name of the
+    qube hosting the USB controller is displayed before the colon in the device
+    name. The string after the colon is the name of the device used within the
+    qube, like so:
 
-(Change `sys-usb` to your desired USB qube.)
+        dom0:sdb1     Cruzer () 4GiB
+        
+        usbVM:sdb1    Disk () 2GiB
 
-You can now use your USB keyboard.
+    **Note:** If your device is not listed here, you may refresh the list by
+    calling (from the qube to which the device is connected):
```

Comment:
I'd clarify that "the qube to which the device is connected" is usbVM in most cases..